src/google-classroom-extractor/tests/api/courses/test_courses_request.py

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from unittest.mock import patch, Mock
+from google_classroom_extractor.api.courses import request_latest_courses_as_df
+
+
+def describe_when_requesting_latest_courses():
+    @patch("google_classroom_extractor.api.courses.call_api")
+    def it_should_use_courses_list_resource(mock_call_api):
+        # arrange
+        resource = Mock()
+        # act
+        request_latest_courses_as_df(resource)
+        # assert
+        assert isinstance(resource.courses.return_value.list, Mock)
```

Comment:
My original thought was that since I had sync integration tests in this area, I would do pure unit tests for the rest since the code was simple. Unfortunately, these functions are so small that unit test granularity lead me to a low-quality tautological test that I'm calling `call_api()` with the correct resource.  The only other thing to test at this level of granularity is that I'm using `json_normalize()`, another low-quality test.

Reply:
OK. I do see your point about the actual work in the function being very trivial - thanks to the great features available to us, like `json_normalize()`. But it could have been more complicated. What if there was something like a mock response with `{'id':1}` and validation that the return is a DataFrame containing that single entry?